### PR TITLE
Shayan/ChartIQ upgrade to V8

### PR DESCRIPTION
**Pull request by shayan-binary-2**

### CODE REVIEW

File: src/chartiq_injections/setMeasure.js [hunk 1]
```diff
@@ -5,26 +5,54 @@ export const overideMeasure = (option) => {
         if (!price1 && price1 !== 0) {
             if (!this.anyHighlighted && this.currentVectorParameters.vectorType === '') this.clearMeasure();
         } else {
+            if (price2 !== false) {
+                let distance =                    Math.round(Math.abs(price1 - price2) * this.chart.roundit)
```
Comment (balakrishna-binary) [inline src/chartiq_injections/setMeasure.js:9]: can this be formatted?

File: src/components/Shade.jsx [hunk 2]
```diff
@@ -9,7 +10,7 @@ const Shade = ({
     setShadeRef,
 }) => (
     <div
-        className={`shade ${className || ''} ${visible ? '' : 'hidden'}`}
+        className={classNames('shade', className || '', { hidden: !visible })}
```
Comment (zaki-binary) [inline src/components/Shade.jsx:13]:
```suggestion
        className={classNames('shade', className, { hidden: !visible })}
```

`classNames` is capable of handling `whitespaces` and `undefined` `classNames`.

File: src/components/ToolbarWidget.jsx [hunk 3]
```diff
@@ -14,7 +15,7 @@ const ToolbarWidget = ({
 
     return (
         <div
-            className={`sc-toolbar-widget sc-toolbar-widget--${position}`}
+            className={classNames('sc-toolbar-widget', `sc-toolbar-widget--${position}`)}
```
Comment (zaki-binary) [inline src/components/ToolbarWidget.jsx:18]:
```suggestion
            className={classNames('sc-toolbar-widget', {`sc-toolbar-widget--${position}`: !!position })}
```

File: src/components/Tooltip.jsx [hunk 4]
```diff
@@ -9,7 +10,14 @@ const Tooltip = ({
     ...props
 }) => (
     <div
-        className={`sc-tooltip sc-tooltip--${position} ${enabled ? 'sc-tooltip--enable' : ''} ${className}`}
+        className={classNames(
+            'sc-tooltip',
+            `sc-tooltip--${position}`,
+            className,
+            {
+                'sc-tooltip--enable': enabled,
```
Comment (zaki-binary) [inline src/components/Tooltip.jsx:18]:
```suggestion
                'sc-tooltip--enable': enabled,
                `sc-tooltip--${position}`: !!position,
```

File: src/components/Tooltip.jsx [hunk 5]
```diff
@@ -9,7 +10,14 @@ const Tooltip = ({
     ...props
 }) => (
     <div
-        className={`sc-tooltip sc-tooltip--${position} ${enabled ? 'sc-tooltip--enable' : ''} ${className}`}
+        className={classNames(
+            'sc-tooltip',
+            `sc-tooltip--${position}`,
```
Comment (zaki-binary) [inline src/components/Tooltip.jsx:15]:
```suggestion
```

File: src/components/Shade.jsx (hunk 2 above)
Comment (shayan-binary-2) [reply src/components/Shade.jsx]: good point. thanks

File: src/components/ToolbarWidget.jsx (hunk 3 above)
Comment (shayan-binary-2) [reply src/components/ToolbarWidget.jsx]: sure

File: src/components/Tooltip.jsx (hunk 4 above)
Comment (shayan-binary-2) [reply src/components/Tooltip.jsx:18]: fixed

File: src/components/Tooltip.jsx (hunk 5 above)
Comment (shayan-binary-2) [reply src/components/Tooltip.jsx]: handled

**Review by zaki-binary — APPROVED** [review]

**Review by zaki-binary — DISMISSED** [review]

**Review by zaki-binary — APPROVED** [review]

**Pull request merged**